consensus/benchmark_test.go

```diff
@@ -0,0 +1,56 @@
+package consensus
+
+import (
+	"bytes"
+	"crypto/rand"
+	"testing"
+)
+
+func nBytes(n int) []byte {
+	buf := make([]byte, n)
+	n, _ = rand.Read(buf)
+	return buf[:n]
+}
+
+func benchmarkWalDecode(b *testing.B, n int) {
+	buf := new(bytes.Buffer)
+	enc := NewWALEncoder(buf)
+
+	data := nBytes(n)
+	enc.Encode(data)
+
+	encoded := buf.Bytes()
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		buf.Reset()
+		buf.Write(encoded)
+		dec := NewWALDecoder(buf)
+		if _, err := dec.Decode(); err == nil {
```

Comment:
this returns an error (`"failed to decode data: %v"`) because you're encoding raw bytes, not TimedWALMessage.

we should probably change `Encode(v interface{})` to `Encode(msg *TimedWALMessage)`

Reply:
Awesome, thank you for the catch @melekes, I'll update it in a second.